scripts/check-swift-format.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftNIO open source project
+##
+## Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+set -euo pipefail
+
+log() { printf -- "** %s\n" "$*" >&2; }
```

Comment:
I am running a shell check locally already but yes we should setup a GH action for it

Reply:
Shell check would've spotted the above quoting error, right?

Reply:
indeed

```
$ echo -e 'files="foo bar"\necho $files' | shellcheck --shell=bash -

In - line 2:
echo $files
     ^----^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
echo "$files"

For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

Reply:
Shell check found two quoting warnings not errors for line 29 and line 33 https://www.shellcheck.net/wiki/SC2086.

Reply:
Right, those are the ones I mean

Reply:
Yeah, we should be shellcheck-clean IMO. The scripts you're cribbing from are IIRC. (I know they trigger Johannes about the capitalized variables, but otherwise).

Reply:
Capitalised variables are bad as they have unknown, unintended side effects. The space of lower case vars is reserved (by POSIX) for app code.

Where we _do want_ side effects (like with `GIT_PAGER=....`) they're great. That's exactly how they're meant to be used:
- UPPER_CASE_STUFF -> side effects in other programs
- lower_case_stuff -> my everyday variables that I do not want to go to other programs

Reply:
I'm not arguing with you here. I have long used uppercase variables in my bash scripts, but I have heard your arguments against them and plan to incorporate them in the future.

FWIW, I don't think the side-effect argument is actually true, unless the variable is `exported`, but I'm still planning on stopping this practice in my own scripts because it helps with local reasoning.

Reply:
Right, for some reason this isn't widely known. But I think it's well documented:
- stackoverflow: https://stackoverflow.com/questions/673055/correct-bash-and-shell-script-variable-capitalization
- posix: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html

And yes, the `export`ed thing is correct (unless you prefix a command/script) but these things happen by accident.  And as you point out, it's just hard to reason about it. If you assign to a variable called `IFS` I don't know if you're meaning to overwrite `IFS` for say `read` or if you want a variable called `IFS`. If you name it `ifs=123` on the other hand I know what you mean.

Reply:
We could also use `local` variables more, I try to do that everywhere except when needing to export vars.

Reply:
Yep, I readily accept that I have fallen into the trap of shouty-case for all my scripting variables. I'm not alone, I know. But I always knew when they were and weren't exported so they _didn't_ affect behaviour, and I've learned to not name my variables to clash with ones that I'll need to affect behaviour. But, like I've already said, I've started to lowercase—we can all grow.